src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1981,42 +2069,50 @@ private static IEnumerable<InstrumentationKind> ParseInstrumentationKinds(string
             return new ResourceDescription(resourceName, fileName, dataProvider, isPublic, embedded, checkArgs: false);
         }
 
-        private static IEnumerable<string> ParseWarnings(string value)
+        private static void ParseWarnings(ReadOnlyMemory<char> value, ArrayBuilder<string> ids)
         {
             value = value.Unquote();
-            string[] values = value.Split(new char[] { ',', ';', ' ' }, StringSplitOptions.RemoveEmptyEntries);
-            foreach (string id in values)
+            var parts = ArrayBuilder<ReadOnlyMemory<char>>.GetInstance();
+            ParseSeparatedStrings(value, s_warningSeparators, removeEmptyEntries: true, parts);
+            foreach (ReadOnlyMemory<char> part in parts)
             {
-                if (string.Equals(id, "nullable", StringComparison.OrdinalIgnoreCase))
+                if (part.Span.Equals("nullable".AsSpan(), StringComparison.OrdinalIgnoreCase))
```

Comment:
Does `.AsSpan()` happen each time inside the loop? Consider moving it outside if so.